src/command_modules/azure-cli-interactive/azclishell/frequency_heuristic.py

```diff
@@ -0,0 +1,63 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import datetime
+import json
+
+from azure.cli.core._config import set_global_config_value
+
+from azclishell.configuration import CONFIGURATION, get_config_dir as shell_config
+
+SHELL_CONFIG = CONFIGURATION
+DAYS_AGO = 28
+ACTIVE_STATUS = 5
+DISPLAY_TIME = 20
+
+
+def day_format(now):
+    """ returns the date format """
+    return now.strftime("%Y-%m-%d")
+
+
+def update_frequency():
+    """ updates the frequency from files """
+    try:
+        with open(os.path.join(shell_config(), SHELL_CONFIG.get_frequency()), 'r') as freq:
+            try:
+                frequency = json.load(freq)
+            except ValueError:
+                frequency = {}
+    except IOError:
```

Comment:
Should catch `OSError` as well.

> In fact, it is hard to think of any situation where OSError should be caught but not IOError, or the reverse.

https://www.python.org/dev/peps/pep-3151/#confusing-set-of-os-related-exceptions